### ISSUE TITLE
Option to use highlightjs for syntax highlighting

### ISSUE THREAD

**Issue by harpocrates** [lightbend/paradox#450]
The API for [`highlightjs`](https://highlightjs.org/) seems to be pretty much the same as `prettyify`. I think both are quite widely used. However,

  * `highlightjs` is actively maintained and developed while `prettyify` is [no longer maintained](https://github.com/googlearchive/code-prettify)
  * `highlightjs` supports more languages

Would it be feasible to add an option for choosing the syntax highlighting library to use?

**Comment by ennru** [comment]
Sure, I would be OK with switching to highlightjs and dropping prettify.

Are you in a position to suggest the necessary changes in a PR?

**Comment by harpocrates** [comment]
> Are you in a position to suggest the necessary changes in a PR?

Not yet - I wanted to check this would be well-received first before moving forward with the actual changes. Are you sure it is OK to drop `prettify`, or should it be kept around as an option (checking as I imagine `highlightjs` will have subtly different syntax definitions and color schemes such that code in existing doc projects like `akka-docs` might end up looking slightly different)?

**Comment by ennru** [comment]
I don't think it is a problem if the syntax colouring slightly changes. highlightjs seems to offer a vast set of predefined designs. Configuring that will be more useful than switching between prettify and highlightjs.

**Comment by harpocrates** [comment]
OK thanks; that definitely lowers the bar for changes I'd have to make/test. I'll aim to open a PR for this in the next week or so.

**Comment by raboof** [comment]
Going straight for highlightjs seems fine to me too.

I'm not sure if the fiddle integration interacts with the syntax highlighting at all - that might be one thing to double-check.

**Comment by pvlugter** [comment]
Switching to highlightjs sounds good to me too. Thanks @harpocrates
